crates/store/src/db/models/queries/notes.rs

```diff
@@ -250,6 +250,29 @@ pub(crate) fn select_note_inclusion_proofs(
     ))
 }
 
+/// Returns the script for a note by its root.
+pub(crate) fn select_note_script_by_root(
+    conn: &mut SqliteConnection,
+    root: Word,
+) -> Result<Option<NoteScript>, DatabaseError> {
+    // SELECT
+    //     root,
+    //     script
+    // FROM
+    //     note_scripts
+    // WHERE
+    //     root = ?1;
+
+    let raw = SelectDsl::select(schema::note_scripts::table, schema::note_scripts::script)
+        .filter(schema::note_scripts::script_root.eq(root.to_bytes()))
+        .get_result::<Vec<u8>>(conn)
+        .optional()?;
+
+    let note_script = raw.map(|bytes| NoteScript::from_bytes(&bytes[..])).transpose()?;
+
+    Ok(note_script)
```

Comment:
Nit/style: should be able to elide some of this

```suggestion
    raw.as_ref().map(NoteScript::from_bytes).transpose().map_err(Into::into)
```